domain/src/main/java/org/oppia/android/domain/translation/TranslationController.kt

```diff
@@ -95,15 +106,36 @@ class TranslationController @Inject constructor(
     }
   }
 
+  /**
+   * Returns List data for [SupportedLanguages] available language definitions from proto asset file.
+   */
+  fun getAllAppLanguageDefinitions(): List<OppiaLanguage> {
+    val supportedLanguages = languageConfigRetriever.loadSupportedLanguages()
+    val supportedLanguageList = arrayListOf<OppiaLanguage>()
+    for (LanguageSupportDefinition in supportedLanguages.languageDefinitionsList) {
+      if (LanguageSupportDefinition.hasAppStringId()) {
+        supportedLanguageList.add(LanguageSupportDefinition.language)
+      }
+    }
+    return supportedLanguageList
+  }
+
   /**
    * Returns a data provider for a [OppiaLocale.DisplayLocale] corresponding to the user's selected
    * language for app strings (see [getAppLanguage]).
    */
   fun getAppLanguageLocale(profileId: ProfileId): DataProvider<OppiaLocale.DisplayLocale> {
     val providerId = APP_LANGUAGE_LOCALE_DATA_PROVIDER_ID
-    return getSystemLanguage().transformAsync(providerId) { systemLanguage ->
-      val language = computeAppLanguage(profileId, systemLanguage)
-      return@transformAsync localeController.retrieveAppStringDisplayLocale(language).retrieveData()
+    val systemLanguageProvider = getSystemLanguage()
+    val appLanguageCacheStore = retrieveCacheStore(profileId)
+
+    return systemLanguageProvider.combineWithAsync(
+      appLanguageCacheStore,
```

Comment:
```suggestion
    return getSystemLanguage().combineWithAsync(
      retrieveCacheStore(profileId),
```

Could be a bit more compact, I think.

Reply:
Yeah for sure nice to compact it like this.